### PR TITLE
JIP-7 Navbar buttons/dropdown, map gets current location, remove location kraft single

### PR DESCRIPTION
**Pull request by srhsrhsrhsrh**
Changes
- When there is no user in the store (it's in the initial state), this is what the user sees in the top right:
image
- When the user is logged in:
image
- What more needs to be done for these buttons and dropdown:
   - Login/Signup dialog
   - Logout action
   - Something to edit account with in the future (not too important)
- Initial location for the map is now your current location - but don't quote me on that (it's telling me I live in Brentwood but I have some sketchy wifi setup so just lmk if it's saying you're somewhere you're not)
- DEVOURED some cheese slices

The commits look like a lot because I used my old navbar branch o.0

### CODE REVIEW

**Comment by srhsrhsrhsrh** [comment]
I NEED TO DEACTIVATE ANOTHER API KEY

File: packages/client/src/pages/common/nav/nav.tsx [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
   },
-  profileButton: {
-    textTransform: "none",
-  }
+  menuButton: {
+    marginLeft: theme.spacing(2),
+  },
 }));
 
+function profileOptions(user: any, classes: any, setModalOpen: any) {
```
Comment (dryu99) [inline packages/client/src/pages/common/nav/nav.tsx:27]: ARE WE USING FUNCTION FUNCTIONS OR ARROW FUNCTIONS AAAAAAAAA

File: packages/client/src/redux/slices/location-slice.ts [hunk 2]
```diff
@@ -1,61 +0,0 @@
-import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
```
Comment (dryu99) [inline packages/client/src/redux/slices/location-slice.ts:1]: oops i thought we were just gonna delete the api service but still have a slice for this to hold state (so other components can access current location if needed e.g. post list). maybe i misunderstood tho and someone else can chime in

**Review by dryu99 — APPROVED** [review]
llllggggttttmmmm

File: packages/client/src/redux/slices/location-slice.ts (hunk 2 above)
Comment (srhsrhsrhsrh) [reply packages/client/src/redux/slices/location-slice.ts:1]: ASHGAJHGDJHF IS IT GONNA BE A SLICE

File: packages/client/src/pages/common/nav/nav.tsx (hunk 1 above)
Comment (srhsrhsrhsrh) [reply packages/client/src/pages/common/nav/nav.tsx:27]: MY BRAIN DEFAULTS TO FUNCTION FUNCTIONS AAAAAAAAAA

File: packages/client/src/redux/slices/location-slice.ts (hunk 2 above)
Comment (dryu99) [reply packages/client/src/redux/slices/location-slice.ts:1]: JUST LEAVE IT AS A SLICE IF ITS NOT TOO MUCH WORK OW LGTM

Comment (srhsrhsrhsrh) [reply packages/client/src/redux/slices/location-slice.ts:1]: I'LL UPDATE TMR

**Pull request merged**